first-react-app/src/components/SubmitForm.tsx

```diff
@@ -0,0 +1,111 @@
+import React from "react";
+import CheckboxInput from "./CheckboxInput";
+import DateInput from "./DateInput";
+import RadioInput from "./RadioInput";
+import SelectInput from "./SelectInput";
+import TextInput from "./TextInput";
+import UploadInput from "./UploadInput";
+
+export interface IFormData {
+  (a: string[]): void;
+}
+
+interface ISubmit extends Object {
+  isFormCorrect: IFormData;
+}
+
+class SubmitForm extends React.Component<ISubmit, { popUpClass: string; popUpText: string }> {
+  textRef: React.RefObject<TextInput>;
+  dateRef: React.RefObject<DateInput>;
+  selectRef: React.RefObject<SelectInput>;
+  checkboxRef: React.RefObject<CheckboxInput>;
+  radioRef: React.RefObject<RadioInput>;
+  uploadRef: React.RefObject<UploadInput>;
+  constructor(props: ISubmit) {
+    super(props);
+    this.state = {
+      popUpClass: "error-hidden",
+      popUpText: "You have errors in form. Please correct them.",
+    };
+    this.textRef = React.createRef();
+    this.dateRef = React.createRef();
+    this.selectRef = React.createRef();
+    this.checkboxRef = React.createRef();
+    this.radioRef = React.createRef();
+    this.uploadRef = React.createRef();
+    this.handleErrors = this.handleErrors.bind(this);
```

Comment:
Since you moved all the ref related logic into individual components, what is the point of keeping them here